feature/experimental/policy/policy_vrf_selection/otg_tests/protocol_dscp_rules_for_vrf_selection_test/protocol_dscp_rules_for_vrf_selection_test.go

```diff
@@ -260,55 +264,72 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
 }
 
 // getIPinIPFlow returns an IPv4inIPv4 *ondatra.Flow with provided DSCP value for a given set of endpoints.
-func getIPinIPFlow(args *testArgs, dstEndPoint ondatra.Endpoint, flowName string, dscp uint8) *ondatra.Flow {
-
-	ethHeader := ondatra.NewEthernetHeader()
-	outerIPHeader := ondatra.NewIPv4Header().WithDSCP(dscp)
-	innerIPHeader := ondatra.NewIPv4Header()
-	innerIPHeader.WithSrcAddress("198.51.100.1")
-	innerIPHeader.DstAddressRange().WithMin("203.0.113.1").WithCount(10000).WithStep("0.0.0.1")
-	flow := args.ate.Traffic().NewFlow(flowName)
-	flow.WithHeaders(ethHeader, outerIPHeader, innerIPHeader)
-	flow.WithSrcEndpoints(args.srcEndPoint)
-	flow.WithDstEndpoints(dstEndPoint)
-	flow.WithFrameRateFPS(100)
-	flow.WithFrameSize(1024)
+func getIPinIPFlow(args *testArgs, src attrs.Attributes, dst attrs.Attributes, flowName string, dscp int32) gosnappi.Flow {
+
+	flow := gosnappi.NewFlow().SetName(flowName)
+	flow.Metrics().SetEnable(true)
+	flow.TxRx().Device().SetTxNames([]string{src.Name + "." + args.iptype}).SetRxNames([]string{dst.Name + "." + args.iptype})
+	ethHeader := flow.Packet().Add().Ethernet()
+	ethHeader.Src().SetValue(src.MAC)
+	outerIPHeader := flow.Packet().Add().Ipv4()
+	outerIPHeader.Src().SetValue(src.IPv4)
+	outerIPHeader.Dst().SetValue(dst.IPv4)
+	outerIPHeader.Priority().Dscp().Phb().SetValue(dscp)
+	innerIPHeader := flow.Packet().Add().Ipv4()
+	innerIPHeader.Src().SetValue("198.51.100.1")
+	innerIPHeader.Dst().SetValue("203.0.113.1")
```

Comment:
In ATE we had an address range for the Dst. I think we don't have that here? Maybe you need innerIPHeader.Dst().SetIncrement(...), but I'm super familiar with OTG

Reply:
sorry, missed it. Added it